x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/index.tsx

```diff
@@ -236,27 +239,36 @@ export const AgentListPage: React.FunctionComponent<{}> = () => {
 
         try {
           setIsLoading(true);
-          const [agentsResponse, totalInactiveAgentsResponse, agentTagsResponse] =
-            await Promise.all([
-              sendGetAgents({
-                page: pagination.currentPage,
-                perPage: pagination.pageSize,
-                kuery: kuery && kuery !== '' ? kuery : undefined,
-                sortField: getSortFieldForAPI(sortField),
-                sortOrder,
-                showInactive,
-                showUpgradeable,
-                getStatusSummary: true,
-                withMetrics: displayAgentMetrics,
-              }),
-              sendGetAgentStatus({
-                kuery: AgentStatusKueryHelper.buildKueryForInactiveAgents(),
-              }),
-              sendGetAgentTags({
-                kuery: kuery && kuery !== '' ? kuery : undefined,
-                showInactive,
-              }),
-            ]);
+          const [
```

Comment:
This `useCallback` hook is getting really big - should we extract this into a dedicated hook that can easily be tested in isolation? I worry that we don't have much coverage on this logic simply becuase we have to test the whole component.